Rust/Rust.sublime-syntax

```diff
@@ -224,80 +207,83 @@ contexts:
-    - match: '[!<>=]=|[<>]'
-      scope: keyword.operator.comparison.rust
-    - match: '='
-      scope: keyword.operator.assignment.rust
-      push: after-operator
-    - match: '&&|\|\||!'
-      scope: keyword.operator.logical.rust
-    - match: '[-+%/*]'
-      scope: keyword.operator.arithmetic.rust
-    - match: '[&|^]|<<|>>'
-      scope: keyword.operator.bitwise.rust
-    - match: '[@~?$#'']'
-      scope: keyword.operator.rust
-    - match: '\b_\b'
-      scope: keyword.operator.rust
-
+  visibility:
+    - match: '\b(pub)\s*(\()'
+      captures:
+        1: storage.modifier.rust
+        2: punctuation.section.group.begin.rust
+      push:
+        - include: comments
+        - match: '\)'
+          scope: punctuation.section.group.end.rust
+          pop: true
+        - match: '(crate|in|self|super)'
+          scope: keyword.other.rust
+        - match: '::'
+          scope: meta.path.rust
+        - match: '{{identifier}}'
+          scope: meta.path.rust
+    - match: '\bpub\b'
+      scope: storage.modifier.rust
 
-  attributes:
-    - match: '(#!?)(\[)'
+  attribute:
+    - match: '(#)\s*(!?)\s*(\[)'
       captures:
         1: punctuation.definition.annotation.rust
-        2: punctuation.section.group.begin.rust
+        2: punctuation.definition.annotation.rust
+        3: punctuation.section.group.begin.rust
       push:
+        # https://github.com/sublimehq/Packages/issues/709#issuecomment-266835130
         - meta_scope: meta.annotation.rust
-        - match: '({{identifier}})(\()'
+        - match: '(?:{{identifier}}\s*::\s*)*({{identifier}})'
+          scope: meta.path.rust
           captures:
             1: variable.annotation.rust
-            2: meta.annotation.parameters.rust meta.group.rust punctuation.section.group.begin.rust
+        - match: '\('
+          scope: meta.annotation.parameters.rust meta.group.rust punctuation.section.group.begin.rust
           push:
             - meta_content_scope: meta.annotation.parameters.rust meta.group.rust
             - match: \)
               scope: meta.annotation.parameters.rust meta.group.rust punctuation.section.group.end.rust
               pop: true
             - include: attribute-call
-        - match: '({{identifier}})'
-          scope: variable.annotation.rust
         - match: '='
-          scope: keyword.operator.rust
+          scope: keyword.operator.assignment.rust
+          set:
+            - meta_content_scope: meta.annotation.rust
+            - include: strings
+            - include: chars
+            - include: numbers
+            - include: comments
+            - match: '\]'
+              scope: meta.annotation.rust punctuation.section.group.end.rust
+              pop: true
+
         - match: '\]'
           scope: punctuation.section.group.end.rust
           pop: true
-        - match: '[ \t]+'
-        - include: strings
+        - include: comments
 
   attribute-call:
     - match: \)
```

Comment:
`attribute-call` is included in line 264, which already defines a pattern for `\)` to pop a context. This makes this one being a duplicate then. Would this pattern probably be better placed before the `-include: atttribute-call` in row 293?